### PR TITLE
Fixes blastdoors/shutters button spam

### PR DESCRIPTION
**Pull request by yoyobatty**
## About The Pull Request
I fixed being able to open and close shutters and have their sound effect being spammed in this one.

## Why It's Good For The Game
Definitely not intended.

## Changelog
:cl: YoYoBatty
fix: You can no longer spam shutters/blastdoors open and closed
/:cl:

### CODE REVIEW

**Comment by Anonmare** [comment]
Wait, you prevented shutters and blast doors being spammed instead of preventing the sound from being spammed or am I misreading? 
Since that's gonna fuck with door crushing as a killing method

**Comment by yoyobatty** [comment]
It's so you can't trigger them when they're already opening/closing which also fixes the sound spam.

**Comment by AnturK** [comment]
Ugh, seems this whole thing was broken in #22179, i feel responsible.

File: code/modules/assembly/doorcontrol.dm [hunk 1]
```diff
@@ -14,13 +14,15 @@
...
-	cooldown = TRUE
 	var/openclose
+	if(cooldown)
+		return
 	for(var/obj/machinery/door/poddoor/M in GLOB.machines)
 		if(M.id == src.id)
 			if(openclose == null || !sync_doors)
 				openclose = M.density
 			INVOKE_ASYNC(M, openclose ? /obj/machinery/door/poddoor.proc/open : /obj/machinery/door/poddoor.proc/close)
+			cooldown = TRUE
```
Comment (AnturK) [inline code/modules/assembly/doorcontrol.dm]: Don't move this after the actual opening just add the missing check.

**Review by AnturK — CHANGES_REQUESTED** [review]
Also this issue applies to all other assemblies in this file.
Better solution here would using the base assembly cooldown system but you'd have to change hardcoded cooldown in there.

**Review by SpaceManiac — APPROVED** [review]

**Pull request merged**